jax/_src/cpu_lib_ops/fused_attention_cpu.py

```diff
@@ -0,0 +1,390 @@
+# Copyright 2025 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any
+import functools
+import json
+import logging
+import numpy as np
+import os
+import platform
+import sys
+
+import jax
+import jax.numpy as jnp
+from jax._src import core
+from jax._src.typing import Array
+from jax._src.interpreters import mlir
+from jax.interpreters import xla
+from jax._src.lib.mlir import ir
+from jax._src.lib.mlir.dialects import hlo
+
+logger = logging.getLogger(__name__)
+
+def _is_dtype_supported(dtype: jnp.dtype, cpuinfo: str):
+   if dtype == jnp.float32:
+      return True
+   elif dtype == jnp.bfloat16:
+      has_bf16 = any(flag in cpuinfo for flag in ["avx512f", "avx_ne_convert", "amx_bf16"])
+      return True
+   elif dtype == jnp.float16:
+      has_f16 = any(flag in cpuinfo for flag in ["avx_ne_convert", "amx_fp16"])
+      has_f16_combined_flags = "avx512_fp16" in cpuinfo and "avx512bw" in cpuinfo
+      return has_f16 or has_f16_combined_flags
+   else:
+      return False
+
+def can_use_cpu_fused_attention(dtype):
+    xla_flags = os.environ.get("XLA_FLAGS", "")
+    flags = xla_flags.split()
+    # XLA sets xla_cpu_use_xnnpack flag to true by default
+    # Currently fused attention op is only supported with oneDNN path
+    # so we need to ensure that xnnpack path is disabled
+    # This can be scaled to support xnnpack path in future if needed
+    if ("--xla_cpu_use_xnnpack=false" not in flags):
```

Comment:
Instead of a negative check, is there a way to turn this into a positive check? As is, it seems like XLA could choose to pick yet a different path from XNNPACK and oneDNN?

E.g., I see that XLA has a `--xla_cpu_use_onednn` knob. Would checking that work?